Remove dependency on dev-haskell/cabal from its DEPENDs

solpeth opened the pull request:
It appears that from `cabal-2.2` onwards, packages such as `mtl` and `parsec` are explicitly depended on. This is causing circular dependencies as per bug #810.

I've removed the packages from `DEPEND` to break the circular dependencies.

Package-Manager: Portage-2.3.47, Repoman-2.3.10

trofi commented:
This change will likely make latest `cabal` unbuildable with older `ghc` versions (as older `ghc` versions don't bundle `mtl`, `parsec` and `text`).

I suggest using `CABAL_FEATURES+=" nocabaldep"` in `cabal`'s `DEPEND`s/`RDEPEND`s instead (would need a comment as well).

https://github.com/gentoo-haskell/gentoo-haskell/issues/810#issuecomment-414112022

solpeth commented:
Okay, I can get onto this in a few hours. Cheers

solpeth commented:
@trofi I've used `nocabaldep` in `mtl`, `parsec` and `text` and removed the dependency on `cabal` from their `DEPEND`s. I've also reverted to removing the test-suite dependencies from `cabal` and restricting tests.
